src/Infrastructure.EntityFramework/AdminConsole/Repositories/OrganizationRepository.cs

```diff
@@ -199,6 +199,14 @@ await dbContext.ApiKeys.Where(ak => ak.ServiceAccount.OrganizationId == organiza
             await dbContext.ServiceAccount.Where(sa => sa.OrganizationId == organization.Id)
                 .ExecuteDeleteAsync();
 
+            var notificationStatuses = from ns in dbContext.NotificationStatuses
+                                       join n in dbContext.Notifications on ns.NotificationId equals n.Id
+                                       where n.OrganizationId == organization.Id
+                                       select ns;
+            dbContext.NotificationStatuses.RemoveRange(notificationStatuses);
```

Comment:
```suggestion
        await dbContext.NotificationStatuses.Where(ns => ns.Notification.OrganizationId == organization.Id).ExecuteDeleteAsync();
```

Can we use `ExecuteDeleteAsync` here?

`.RemoveRange()` Will query for all notification status that match the filter, load it into application memory, then submit a delete statement for each.

More information here https://learn.microsoft.com/en-us/ef/core/saving/execute-insert-update-delete#executedelete

Reply:
Nice, i will make that change.
It's still blurry to me where and how EF code is translated to SQ and which parts will it optimize. I will remember this one in the future.

Reply:
Fixed in 1c48155